src/utils/merge.js

```diff
@@ -19,9 +22,40 @@ export default function mergeObjects (dest, ...sources) {
                 }
             }
             // Overwrite the previous destination value if the source property is: a scalar (number/string),
-            // an array, or a null value
+            // an array (unless using array merge operator below), or a null value
             else if (value !== undefined) {
-                dest[key] = value;
+                let merge_index = is_array && value.indexOf(array_merge_op); // special array merge syntax
+                if (is_array && dest[key] != null && merge_index > -1) {
+                    // If the source value includes the '...' operator, the previous destination value will be
+                    // merged into the source value in its place.
+                    //
+                    // Example 1: append values
+                    //   dest: [1, 2 3]
```

Comment:
Probably want a comma in `2 3` - same in the next example.

Reply:
whoops